draft-yasskin-dispatch-bundled-exchanges.md

```diff
@@ -0,0 +1,675 @@
+   an error, return an error.
+
+1. Let `requests` be an initially-empty map ({{INFRA}}) from HTTP requests
+   ({{FETCH}}) to structs ({{INFRA}}) with items named "offset" and "length".
+
+1. For each `cbor-http-request`/\[`offset`, `length`] triple in `index`:
+   1. Let `headers`/`pseudos` be the result of converting `cbor-http-request` to
+      a header list and pseudoheaders using the algorithm in {{cbor-headers}}.
+      If this returns an error, return that error.
+   1. If `pseudos` does not have keys named ':method' and ':url', or its size
+      isn't 2, return an error.
+   1. If `pseudos[':method']` is not 'GET', return an error.
+
+      Note: This could probably support any cacheable (Section 4.2.3) of
+      {{!RFC7231}}) and safe (Section 4.2.1 of {{!RFC7231}}) method, matching
+      PUSH_PROMISE (Section 8.2 of {{?RFC7540}}), but today that's only HEAD and
+      GET, and HEAD can be served as a transformation of GET, so this version of
+      the specification keeps the method simple.
+   1. Let `parsedUrl` be the result of parsing ({{URL}}) `pseudos[':url']` with
+      no base URL.
+   1. If `parsedUrl` is a failure, its fragment is not null, or it includes
+      credentials, return an error.
+   1. Let `http-request` be a new request ({{FETCH}}) whose:
+      * method is `pseudos[':method']`,
+      * url is `parsedUrl`,
+      * header list is `headers`, and
+      * client is null.
+
+   1. Let `streamOffset` be `sectionsStart + section-offsets\["responses"].offset
+      + offset`. That is, offsets in the index are relative to the start of the
+      "responses" section.
+   1. If `offset + length` is greater than
+      `section-offsets\["responses"].length`, return an error.
+   1. Set `requests`\[`http-request`] to a struct whose "offset" item is
+      `streamOffset` and whose "length" item is `length`.
+
+1. Set `metadata`'s "requests" item to `requests`.
+
+### Parsing the manifest section {#manifest-section}
+
+The "manifest" section records a single URL identifying the manifest of the
+bundle. The bundle can contain multiple resources at this URL, and the client is
+expected to content-negotiate for the best one. For example, a client might
+select the one with an `accept` header of `application/manifest+json` and an
+`accept-language` header of `es-419`.
+
+~~~ cddl
+manifest = text
+~~~
+
+To parse the manifest section, given its `sectionContents` and the `metadata`
+struct to fill in, the parser MUST do the following:
+
+1. Let `urlString` be the result of parsing `sectionContents` as a CBOR item
+   matching the above `manifest` rule ({{parse-cbor}}. If `urlString` is an
+   error, return that error.
+
+1. Let `url` be the result of parsing ({{URL}}) `urlString` with no base URL.
+
+1. If `url` is a failure, its fragment is not null, or it includes credentials,
+   return an error.
+
+1. Set `metadata`'s "manifest" item to `url`.
+
+### Parsing the critical section {#critical-section}
```

Comment:
Wasn't fully unsure what could come here?

Reply:
If we add more sections in the future, and the publisher wants to make sure parsers don't just ignore one of those sections because they don't understand it, this section points out the ones not to skip.

I'm not certain this is the right way to do this. PNG does it with a particular capitalization in the section names: https://tools.ietf.org/html/rfc2083#page-13